### PR TITLE
Add search templates

### PR DESCRIPTION
**Pull request by bobbyhubbard**
Added ability to search with templates (inline or stored) resolving #240.

### CODE REVIEW

**Comment by kramer** [comment]
Appreciate the effort, it would be even more complete with an integration test, so do you mind adding one? :)

**Comment by bobbyhubbard** [comment]
Got the unit test missed the integration test. Happy to.
________________________________________

**Comment by bobbyhubbard** [comment]
So I cannot get the integration tests to recognize mustache as a legit script_lang. I've tested it on 2.0 and 2.1 clusters directly and this works fine. However, the integration test config has something goofy where any reference to a mustache template (whether inline or id) fails for mustache not supported. 

`"script_lang not supported [mustache]"`

mustache is built-in and sandboxed and I dont see anywhere in the integration test config where it is explicitly turned off. I've also tried manually setting the script.* settings to no avail. 

@kramer Any recommendation as to why it might be rejecting mustache?

**Comment by kramer** [comment]
I will take a deeper look but my initial guess is mustache is expected as an additional dependency just like groovy.

**Comment by bobbyhubbard** [comment]
You were right. Somewhere along the way (since 2.0?) they stopped shading most dependencies. Which is cool. I just didn't realize it :) Anyway, I'll continue flushing out these integration tests. Thanks for the recommendation.
________________________________________

**Comment by bobbyhubbard** [comment]
Integration tests added.. Thanks for your help @kramer

File: pom.xml [hunk 1]
```diff
@@ -74,6 +74,7 @@
         <hamcrest.version>1.3</hamcrest.version>
         <randomizedtesting.version>2.1.16</randomizedtesting.version>
         <groovy.version>2.4.4</groovy.version>
+        <mustache.version>0.8.18</mustache.version>
```
Comment (kramer) [inline pom.xml]: Better use the [same version as Elasticsearch 2.0](https://github.com/elastic/elasticsearch/blob/2.0/pom.xml#L308) which is `0.8.13`.

**Comment by kramer** [comment]
Thanks @bobbyhubbard !

**Comment by bobbyhubbard** [comment]
@kramer Checking in. Any idea if this pull will be included in your next jest release?

**Comment by kramer** [comment]
@bobbyhubbard It's merged to master so yes of course it'll be included, don't know when though.

**Pull request merged**